src/test/scala/tech/cryptonomic/conseil/tezos/PlatformDiscoveryOperationsTest.scala

```diff
@@ -163,4 +171,83 @@ class PlatformDiscoveryOperationsTest
     }
   }
 
+  "listAttributeValues" should {
+
+    "return list of values of kind attribute of Fees without filter" in {
+      val avgFee = AverageFees(1, 3, 5, Timestamp.valueOf(LocalDateTime.of(2018, 11, 22, 12, 30)), "example1")
+
+      Await.result(dbHandler.run(TezosDatabaseOperations.writeFees(List(avgFee))), Duration.Inf)
+      dbHandler.run(
+        DBIO.sequence(
+          PlatformDiscoveryOperations.verifyAttributesAndGetQueries("fees", "kind", None)
+        )
+      ).futureValue.flatten shouldBe List("example1")
+    }
+
+    "return throw an exeption when asked for medium attribute of Fees without filter - numeric attributes should not be displayed" in {
+      val avgFee = AverageFees(1, 3, 5, Timestamp.valueOf(LocalDateTime.of(2018, 11, 22, 12, 30)), "example1")
+
+      Await.result(dbHandler.run(TezosDatabaseOperations.writeFees(List(avgFee))), Duration.Inf)
+
+      val thrown = intercept[Exception] {
+        dbHandler.run(
+          DBIO.sequence(
+            PlatformDiscoveryOperations.verifyAttributesAndGetQueries("fees", "medium", None)
+          )
+        ).futureValue
+      }
+      thrown.getMessage shouldBe "The future returned an exception of type: java.util.NoSuchElementException, with message: Action.withFilter failed."
+
+    }
+
+    "return empty list when trying to sql inject" in {
+      val avgFee = AverageFees(1, 3, 5, Timestamp.valueOf(LocalDateTime.of(2018, 11, 22, 12, 30)), "example1")
+
+      Await.result(dbHandler.run(TezosDatabaseOperations.writeFees(List(avgFee))), Duration.Inf)
+      // That's how the SLQ-injected string will look like:
+      // SELECT DISTINCT kind FROM fees WHERE kind LIKE '%'; DELETE FROM fees WHERE kind LIKE '%'
+      val maliciousFilter = Some("'; DELETE FROM fees WHERE kind LIKE '")
+
+      dbHandler.run(
+        DBIO.sequence(
+          PlatformDiscoveryOperations.verifyAttributesAndGetQueries("fees", "kind", maliciousFilter)
+        )
+      ).futureValue.flatten shouldBe List.empty
+
+      dbHandler.run(Tables.Fees.length.result).futureValue shouldBe 1
+
+    }
+    "check if filtering works correctly" in {
```

Comment:
This might be better phrased, the subject is "listAttributeValues"

The test case could read like, e.g. "listAttributeValues" should "correctly apply the filter" in ...

Reply:
changed